### ISSUE TITLE
Turbulence agent job assumes hostname returns BOSH agent ID

### ISSUE THREAD

**Issue by iainsproat** [cppforlife/turbulence-release#22]
This is not the case with xenial stemcells. Ideally, it should look up the agent id in `/var/vcap/bosh/settings.json`.

This causes turbulence to not work when the agent is on xenial stemcells.
This is the error that we get from the API:

```
Timed out waiting for agent 'fc9fd54b-765a-4268-bc20-95597ead9110' to consume tasks
```

Lines of code causing this problem: https://github.com/cppforlife/turbulence-release/blob/f6209362ce5f3a686913fee32e462a89cfef8b14/jobs/turbulence_agent/templates/ctl.erb#L14-L16

cc @akshaymankar

**Comment by iainsproat** [comment]
Fixed in BOSH stemcell [v170.13](https://github.com/cloudfoundry/bosh-linux-stemcell-builder/releases/tag/ubuntu-xenial%2Fv170.13)

**Comment by kfkonrad** [comment]
I encountered this exact bug that's been "fixed" in v170.13 with v250.17 so I think CloudFoundry might just have reintroduced it in a later version.

IMO https://github.com/cppforlife/turbulence-release/pull/23 is worth merging.
It's cleaner to have a solution that is independent of the behavior of a command like `hostname` (that could return an arbitrary value) and to extract the data from `/var/vcap/bosh/settings.json`.
